addon/components/infinity-loader.js

```diff
@@ -127,17 +129,17 @@ const InfinityLoaderComponent = Component.extend({
     onExit(instance.didExitViewport.bind(instance));
   },
 
-  willDestroyLoader(_element, [instance]) {
-    instance._cancelTimers();
+  willDestroy() {
+    this._cancelTimers();
 
-    get(instance, 'infinityModelContent')
+    get(this, 'infinityModelContent')
       .then((infinityModel) => {
-        infinityModel.off('infinityModelLoaded', instance, instance._loadStatusDidChange.bind(instance));
+        infinityModel.off('infinityModelLoaded', this, this._loadStatusDidChange.bind(this));
       });
 
-    instance.removeObserver('infinityModel', instance, instance._initialInfinityModelSetup);
-    instance.removeObserver('hideOnInfinity', instance, instance._loadStatusDidChange);
-    instance.removeObserver('reachedInfinity', instance, instance._loadStatusDidChange);
+    this.removeObserver('infinityModel', this, this._initialInfinityModelSetup);
+    this.removeObserver('hideOnInfinity', this, this._loadStatusDidChange);
+    this.removeObserver('reachedInfinity', this, this._loadStatusDidChange);
```

Comment:
👋 Was this change necessary or could we still use the `will-destroy` modifier?

Reply:
Thinking out loud...

If the component hides, do we really want to cleanup?  What does it mean to be not visible?  After implementing our own isVisible, we are just using CSS but not actually tearing down the component.

Reply:
:wave: Yes it was necessary for the second bug mentioned in the OP: `will-destroy` is triggered when setting `isVisible` to `false` which removes the <div> its attached to, which isn't the same thing as the component being destroyed since it can be unhidden again later. Conflating them leaves you in a broken state when it was unhidden since these listeners were missing.